pkg/resource/api.go

```diff
@@ -91,7 +91,7 @@ func NewAPIManagedConnectionPropagator(c client.Client, t runtime.ObjectTyper) *
 }
 
 // PropagateConnection details from the supplied resource to the supplied claim.
-func (a *APIManagedConnectionPropagator) PropagateConnection(ctx context.Context, cm Claim, mg Managed) error {
+func (a *APIManagedConnectionPropagator) PropagateConnection(ctx context.Context, cm Target, mg Managed) error {
```

Comment:
I guess renaming `cm Target` to something like `tr Target` or `t Target` would read better.

Reply:
Yep, if we keep this implementation, I definitely think the variable name should be representative of the most narrow accepted interface.